back/src/oidc-client/oidc-client.saver.ts

```diff
@@ -0,0 +1,34 @@
+import { InjectRepository } from '@nestjs/typeorm';
+import { CreateOidcClientDto } from './oidc-client.dto';
+import { OidcClient } from 'src/entities/oidc-client.entity';
+import { Repository } from 'typeorm';
+
+export class OidcClientSaver {
+  constructor(
+    @InjectRepository(OidcClient)
+    private oidcClientRepository: Repository<OidcClient>
+  ) {}
+
+  private hydrate = (dto: CreateOidcClientDto): OidcClient => {
+    const oidcClient = new OidcClient();
```

Comment:
remove this line outside this function, pass the oidcClient in the params of the function. The responsability of hydrate is not to create a new OidcClient each time, only to add data to an existent one.